samples/sensor/kx022/src/main.c

```diff
@@ -0,0 +1,311 @@
+	static unsigned int motion_cnt;
+
+	fetch_and_display(dev);
+	motion_display(dev);
+	if (++motion_cnt > 5) {
+		motion_test_done = 1;
+		motion_cnt = 0;
+	}
+}
+static void slope_handler(const struct device *dev, struct sensor_trigger *trig)
+{
+	static unsigned int slope_cont ;
+
+	fetch_and_display(dev);
+	tilt_position_display(dev);
+	if (++slope_cont > 5) {
+		slope_test_done = 1;
+		slope_cont = 0;
+	}
+}
+#endif
+
+static void test_polling_mode(const struct device *dev)
+{
+	int32_t remaining_test_time = MAX_TEST_TIME;
+
+	printf("\n");
+	printf("\t\tAccelerometer: Poling test Start\r\n");
+	do {
+		fetch_and_display(dev);
+		/* wait a while */
+		k_msleep(SLEEPTIME);
+
+		remaining_test_time -= SLEEPTIME;
+	} while (remaining_test_time > 0);
+}
+
+static void test_trigger_mode(const struct device *dev)
+{
+#if 0
+	struct sensor_value accel[9];
+#endif
+	struct sensor_value val;
+	struct sensor_trigger trig;
+	uint8_t rc;
+
+	trig.type = SENSOR_TRIG_NEAR_FAR;
+	trig.chan = SENSOR_CHAN_ALL;
+
+	rc = sensor_trigger_set(dev, &trig, slope_handler);
+	printf("\n");
+	printf("\t\tAccelerometer: Tilt Position trigger test Start\r\n");
+
+#if 0
+	rc = sensor_attr_get(dev, SENSOR_ATTR_CONFIGURATION, DEVICE_HANDLE_NULL, accel);
+	rc = sensor_attr_get(dev, SENSOR_CHAN_NEAR_FAR, DEVICE_HANDLE_NULL, accel);
+#endif
+
+	while (slope_test_done == 0) {
+		k_sleep(K_MSEC(200));
+	}
+	slope_test_done = 0;
+	printf("\n");
+	printf("\t\tAccelerometer: Tilt Position trigger test finished\r\n");
+
+	// stop trigger tilt interrupt but tilt function also working
+	// tilt function working mean that KX022_REG_INS2 have new tilt data
+	// disable the tilt function reference kx022.c reference KX022_accel_angle_set function line 311
+	val.val1 = 0;
+	rc = sensor_attr_set(dev, SENSOR_CHAN_NEAR_FAR, SENSOR_ATTR_SLOPE_TH, &val);
+
+	trig.type = SENSOR_TRIG_FREEFALL;
+	trig.chan = SENSOR_CHAN_ALL;
+
+	rc = sensor_trigger_set(dev, &trig, motion_handler);
+	printf("\n");
+	printf("\t\tAccelerometer: Motion  trigger test Start\r\n");
+
+#if 0
+	rc = sensor_attr_get(dev, SENSOR_ATTR_CONFIGURATION, DEVICE_HANDLE_NULL, accel);
+	rc = sensor_attr_get(dev, SENSOR_CHAN_FREE_FALL, DEVICE_HANDLE_NULL, accel);
+#endif
+
+	while (motion_test_done == 0) {
+		k_sleep(K_MSEC(200));
+	}
+	motion_test_done = 0;
+	printf("\n");
+	printf("\t\tAccelerometer: Motion trigger test finished\r\n");
+
+	// stop trigger motion detect interrupt but motion detect function also working
+	// tilt function working mean KX022_REG_INS2 have new motion data
+	// disable the motion detect function reference kx022.c reference KX022_accel_angle_set function line 246
+	val.val1 = 0;
+	rc = sensor_attr_set(dev, SENSOR_CHAN_FREE_FALL, SENSOR_ATTR_WUFF_TH, &val);
+
+	k_sleep(K_MSEC(500));
+}
+
+void main(void)
+{
+#if 0
+	uint8_t rc;
+	struct sensor_value accel[9];
+	struct sensor_value val;
+#endif	
+	const struct device *sensor = device_get_binding(DT_LABEL(DT_INST(0, kionix_kx022)));
+
+	if (sensor == NULL) {
```

Comment:
You need to use `device_is_ready()` to check the device initialization status is OK.